data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceGrpcService.java

```diff
@@ -60,16 +90,45 @@ public void export(ExportTraceServiceRequest request, StreamObserver<ExportTrace
             return;
         }
 
+        Collection<Span> spans = new ArrayList<>();
+        payloadSizeSummary.record(request.getSerializedSize());
+
+        try {
+            spans = OTelProtoCodec.parseExportTraceServiceRequest(request);
+        } catch (Exception e) {
+            LOG.error("Failed to parse the request content [{}] due to:", request, e);
+            badRequestsCounter.increment();
+            responseObserver.onError(Status.INVALID_ARGUMENT.withDescription(e.getMessage()).asException());
+            return;
+        }
+
+        final List<Record<Span>> records = spans.stream()
+                .map(Record::new)
+                .collect(Collectors.toList());
+
         try {
-            buffer.write(new Record<>(request), bufferWriteTimeoutInMillis);
+            buffer.writeAll(records, bufferWriteTimeoutInMillis);
+            successRequestsCounter.increment();
             responseObserver.onNext(ExportTraceServiceResponse.newBuilder().build());
             responseObserver.onCompleted();
-        } catch (TimeoutException e) {
-            LOG.error("Buffer is full, unable to write");
-            requestTimeoutCounter.increment();
-            responseObserver
-                    .onError(Status.RESOURCE_EXHAUSTED.withDescription("Buffer is full, request timed out.")
-                            .asException());
+        } catch (Exception e) {
```

Comment:
I'd rather see specific catch statements for each Exception rather than an if-else in the exception handling. I'd prefer that even if the `LOG.error` is copied across all.

But, I wouldn't reject the PR over that.

Reply:
This generic Exception has been tightly coupled to the exception thrown by writeAll API in the buffer. I will open a separate issue to track the enhancement from throwing generic exception to specific exceptions in buffer.

Reply:
#882

Reply:
My comment was more on the use of `if(e instanceof SpecificException)`. I'd rather: `catch(SpecificException e)`.